packages/back-end/src/routers/saved-queries/saved-queries.controller.ts

```diff
@@ -361,20 +431,32 @@ export async function postGenerateSQL(
       const columnsDescription = value.columns
         .map((column) => `${column.columnName} (${column.dataType})`)
         .join(", ");
-      return `Database: ${value.databaseName}, Table: ${value.tableName}, Schema: ${value.tableSchema}, Columns: [${columnsDescription}]`;
+      return `Database: ${value.databaseName}, Table: ${value.tableName}, Schema name: ${value.tableSchema}, Columns: [${columnsDescription}]`;
     }
   ).join("\n");
 
-  const instructions =
+  let instructions =
     "You are a data analyst and SQL expert.\n" +
     "Generate a SQL query to answer the provided question inputted. The resultant query should be ONLY based on the provided context of tables and structure, and return only valid SQL that is executable on the specified data source." +
+    "\n\nThe database is a " +
+    datasource.type +
+    " database. Be sure to make queries valid for that database type." +
     "\n\nTable structure: " +
     "\n" +
     schemasString +
     "\n\nInput: " +
     input +
-    "\n\nKeep in mind that the table names may be sharded, so you should use the generic table names for tables that have wild cards, such as 'events_*'. If you are querying sharded tables, and you're asked for a date range, you should use something like: \n((_TABLE_SUFFIX BETWEEN '{{date startDateISO \"yyyyMMdd\"}}' AND '{{date endDateISO \"yyyyMMdd\"}}') OR\n" +
-    "   (_TABLE_SUFFIX BETWEEN 'intraday_{{date startDateISO \"yyyyMMdd\"}}' AND 'intraday_{{date endDateISO \"yyyyMMdd\"}}'))\n\n when constructing the query to make sure it's fast and takes use of the sharding. If you do this code, be sure to replace the dates with the correct range.\n";
+    "\n\n" +
+    "Use date ranges if possible and it makes sense with the input. If the input requests a date range, like 'in the past month', use the current date: " +
+    new Date().toISOString() +
+    " as the starting point.";
+
+  if (datasource.type === "bigquery") {
+    instructions +=
+      "\n\nBe sure to use the schema name in the query, ie: databaseName.schemaName.tableName.\n" +
+      "Keep in mind that the table names may be sharded, so you should use the generic table names for tables that have wild cards, such as 'events_*'. If you are querying sharded tables, and you're asked for a date range, you should use something like: \n((_TABLE_SUFFIX BETWEEN '{{date startDateISO \"yyyyMMdd\"}}' AND '{{date endDateISO \"yyyyMMdd\"}}') OR\n" +
+      "   (_TABLE_SUFFIX BETWEEN 'intraday_{{date startDateISO \"yyyyMMdd\"}}' AND 'intraday_{{date endDateISO \"yyyyMMdd\"}}'))\n\n when constructing the query to make sure it's fast and takes use of the sharding. If you do this code, be sure to replace the dates with the correct range.\n";
```

Comment:
The `{{...}}` macros won't work in the SQL Explorer.  The example here should just hardcode dates

```sql
_TABLE_SUFFIX BETWEEN '20250101' AND '20250131'
```

And I would remove the `intraday_` part since it's confusing.